clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -619,11 +838,36 @@ private CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> chainFuture(fi
         public String toString() {
             return "OffsetFetchRequestState{" +
                     "requestedPartitions=" + requestedPartitions +
-                    ", requestedGeneration=" + requestedGeneration +
+                    ", memberId=" + memberId.orElse("undefined") +
+                    ", memberEpoch=" + (memberEpoch.isPresent() ? memberEpoch.get() : "undefined") +
```

Comment:
What about this:

```suggestion
                    ", memberEpoch=" + memberEpoch.orElse("undefined") +
```

Actually, it's similar to what is done above that line.

Reply:
I had to go that way because the orElse expects a param of the same type as the value (memberEpoch.orElse("undefined") does not compile). I truly don't want to print any epoch in that case, just a string.